common/network-common/src/main/java/org/apache/spark/network/util/TransportConf.java

```diff
@@ -265,6 +265,16 @@ public boolean saslServerAlwaysEncrypt() {
     return conf.getBoolean("spark.network.sasl.serverAlwaysEncrypt", false);
   }
 
+  /**
+   * Flag indicating whether to share the pooled ByteBuf allocators between the different Netty
+   * channels. If enabled then only two pooled ByteBuf allocators are created: one where caching
+   * is allowed (for transport servers) and one where not (for transport clients).
+   * When disabled a new allocator is created for each transport servers and clients.
+   */
+  public Boolean sharedByteBufAllocators() {
+    return conf.getBoolean("spark.network.sharedByteBufAllocators.enabled", true);
```

Comment:
Is there any downside to this change? do we even need to let people turn it off? an escape valve is OK if there's some marginal risk, but otherwise when would someone disable it?

Reply:
Yes, lower Netty memory usage is the goal of this change. And the you are right regarding the config value as well: it is there for safety reason.

I am surprised by @duhanmin 's post, so after the Holiday season I plan to run some standalone tests with Netty metrics on and compare the results (with the config on and off).

Reply:
@attilapiros 
We are using Structured Streaming, mainly to do some json parsing calculations, write MYsql tasks, this problem will occur after a long time, while processing 5 topics, each task 3 partitions, there are about 4 data per second , each data size is about 5MB, the startup script:
Spark-submit \
--class ....... \
--driver-memory 3g \
--executor-memory 2g \
--executor-cores 6 \
--num-executors 3 \
.......

Reply:
And you are getting outOfMemory when spark.network.sharedByteBufAllocators.enabled is true and there is no problem when spark.network.sharedByteBufAllocators.enabled is false?

Reply:
I'm not clear the comment above was using this change.
I'm still not sure if there's a meaningful case where you'd not enable sharing, if it should always be a win and we don't know of any problems it causes (i.e. tests pass). I'm not against leaving the flag in, just adds to the number of configs we document, maintain, etc. I wouldn't mind removing it too, as I'm not sure when a user could decide this was something they don't want.

For `spark.network.sharedByteBufAllocators.io.preferDirectBufs`, how about just using `spark.network.io.preferDirectBufs` which exists already? again same question... when would you want these set differently?

Reply:
I think it's good to keep the option to disable this, just in case. We don't need to necessarily document it, but leave it there as an escape hatch if something unexpected shows up.

+1 on sharing the direct buf config, though.

Also, this method should return `boolean`.

Reply:
@srowen I understand your concern fully but I would prefer to keep `spark.network.sharedByteBufAllocators.enabled` at least for a short time for safety reasons.

Regarding `spark.network.sharedByteBufAllocators.io.preferDirectBufs` unfortunately I cannot use`spark.network.io.preferDirectBufs` as it does not exists this way but there are separate parameters with the `.io.preferDirectBufs` suffix for each modules, like `spark.shuffle.io.preferDirectBufs`, `spark.files.io.preferDirectBufs` and `spark.rpc.io.preferDirectBufs`.

For details see the [getConfKey](https://github.com/apache/spark/blob/2f328e842948e97ab53a0d8003fd2e9b54bcfd3b/common/network-common/src/main/java/org/apache/spark/network/util/TransportConf.java#L78) method which concatenates the suffix via the initialisation of the key [SPARK_NETWORK_IO_PREFERDIRECTBUFS_KEY](https://github.com/apache/spark/blob/2f328e842948e97ab53a0d8003fd2e9b54bcfd3b/common/network-common/src/main/java/org/apache/spark/network/util/TransportConf.java#L54).

Reply:
OK, leave in `spark.network.sharedByteBufAllocators.enabled` but don't document it. For the other, yes I know, but why not `spark.network.io.preferDirectBufs`?

Reply:
Ok I will rename `spark.network.sharedByteBufAllocators.io.preferDirectBufs` to  `spark.network.io.preferDirectBufs` and remove the documentation of the `spark.network.sharedByteBufAllocators.enabled`.

Reply:
After merging #23416, could you use `ConfigEntry`?

Reply:
You mean only "spark.network.io.preferDirectBufs" and "spark.network.sharedByteBufAllocators.enabled"? 
There are several other configs  here and I checked #23416 and it does not touch TransportConf.

Reply:
I believe he means it should go in Network.scala. @kiszk for consistency, what about merging this then updating https://github.com/apache/spark/pull/23416 which you have open right now?